ultravox/training/config_base.py

```diff
@@ -2,115 +2,189 @@
 import datetime
 import logging
 import os
-import re
 import sys
 from pathlib import Path
 from typing import Any, Dict, List, Optional
 
 import simple_parsing
-import torch
 
-from ultravox.data import dataset_config
 from ultravox.data import datasets
 from ultravox.model import ultravox_config
+from ultravox.utils import device_helpers
+from ultravox.utils import string_helpers
 
 
 @dataclasses.dataclass
 class TrainConfig:
-    data_sets: List[str]
-    val_sets: List[str]
-    # language model to use
-    text_model: str
-    # audio encoder model to use
-    audio_model: str
-
-    # The data_dicts field complements data_sets, allowing for the inclusion of
-    # new datasets in the config.
-    #
-    # Due to simple_parsing's lack of support for containers of dataclass types,
-    # we first parse the data_dicts as a list of dictionaries. After parsing,
-    # we convert these dictionaries to DataDictConfig objects using Pydantic
-    # to enforce type constraints and validation, in the __post_init__ method.
-    data_dicts: Optional[List[Dict[str, Any]]] = None
+    """Configuration class for training settings."""
+
+    text_model: Optional[str] = None
+    """Language model to use; could be a huggingface model id, wandb path, or local path."""
+
+    audio_model: Optional[str] = None
+    """Audio encoder model to use; could be a huggingface model id, wandb path, or local path."""
```

Comment:
Why use the doc style instead of the comment style?
I find the comments that come after quite confusing.
`simple_parsing` can use both.